mindmeld/components/entity_resolver.py

```diff
@@ -256,32 +305,44 @@ def _trim_results(self, results, top_n):
             _id = result["id"] if "id" in result else result["cname"]
             if _id not in best_results or result["score"] > best_results[_id]["score"]:
                 best_results[_id] = result
-
-        # sort as per top_n requirement using the score field
         results = [*best_results.values()]
+
+        # Obtain upto top_n docs and sort them as final result
         n_scores = len(results)
-        if n_scores < top_n:
-            if top_n != 20:
-                # log only if a value different from default value is specified
-                msg = f"Retrieved only {len(results)} entity resolutions instead of asked " \
-                      f"number {top_n} for entity type {self.type}"
-                logger.info(msg)
-            return results
-        # else, n_scores >= top_n
```

Comment:
Bug. If there are less than top_n resolutions, they weren't sorted before returning. Fixed this.